### PR TITLE
[FLINK-27098] hotfix: Fixes the createFlinkDepInformerEventSource namespace scope

### PR DESCRIPTION
**Pull request by haoxins**
The error will occur if we use the operator with namespaced scope.

```
[WARN ] Error starting org.apache.flink.kubernetes.operator.controller.FlinkSessionJobController$1@96a75da
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://10.16.0.1/apis/flink.apache.org/v1alpha1/namespaces/flink-operator/flinkdeployments. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked. flinkdeployments.flink.apache.org is forbidden: User "system:serviceaccount:flink-operator:flink-operator" cannot list resource "flinkdeployments" in API group "flink.apache.org" in the namespace "flink-operator".
```

### CODE REVIEW

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkSessionJobController.java [hunk 1]
```diff
@@ -162,8 +162,19 @@ public DeleteControl cleanup(FlinkSessionJob sessionJob, Context context) {
...
             String name) {
+        var sharedIndexInformer =
+                ALL_NAMESPACE.equals(name)
+                        ? kubernetesClient
+                                .resources(FlinkDeployment.class)
+                                .inAnyNamespace()
+                                .runnableInformer(0)
+                        : kubernetesClient
+                                .resources(FlinkDeployment.class)
+                                .inNamespace(name)
```
Comment (Aitozi) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkSessionJobController.java:173]: I think we could pass the informer as the first parameter of `createFlinkDepInformerEventSource` like the `OperatorUtils#createJmDepInformerEventSource`. By this, we do not have to assume the name is the namespace

**Review by Aitozi — COMMENTED** [review]
@haoxins Thanks for your fix, I left a minor comment

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkSessionJobController.java (hunk 1 above)
Comment (haoxins) [reply flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkSessionJobController.java]: updated

**Review by Aitozi — APPROVED** [review]
LGTM

**Review by wangyang0918 — APPROVED** [review]
+1

I will merge this PR after manual verification. @Aitozi It will be great if we could introduce an e2e test for session job after all the session job related work.

**Comment by Aitozi** [comment]
> +1
> 
> I will merge this PR after manual verification. @Aitozi It will be great if we could introduce an e2e test for session job after all the session job related work.

Yes, I plan to add the e2e test after finishing the basic spec changing support (I will open the PR asap).

**Pull request merged**